### PR TITLE
feat(api-client): better accessibility in the API client modal

### PR DESCRIPTION
**Pull request by hwkr**
Probably not prefect but should be a big improvement to accessibility in the API Client modal. Please test and make sure everything is still working as expected 🙏

### CODE REVIEW

**Comment by amritk** [comment]
Wanna get this fixed up and ping me and I'll test/review

**Comment by hwkr** [comment]
@amritk This PR should be good to go!

File: packages/api-client/src/components/DataTable/DataTableInput.vue [hunk 1]
```diff
@@ -117,7 +117,7 @@ const handleDropdownMouseUp = () => {
           @update:modelValue="emit('update:modelValue', $event)" />
         <div
           v-if="required"
-          class="absolute centered-y right-0 pt-px pr-2 text-xxs text-c-3 bg-b-1 shadow-[-8px_0_4px_var(--scalar-background-1)] opacity-100 duration-150 transition-opacity peer-focus:opacity-0">
+          class="absolute centered-y right-2 pt-px text-xxs text-c-3 bg-b-1 shadow-[-8px_0_4px_var(--scalar-background-1)] opacity-100 duration-150 transition-opacity peer-has-[:focus-visible]:opacity-0">
```
Comment (amritk) [inline packages/api-client/src/components/DataTable/DataTableInput.vue:120]: no need to change this one as it was from before, but when tw starts getting this crazy its a good time to use a class!

File: packages/api-client/src/views/Request/RequestSidebar.vue [hunk 2]
```diff
@@ -103,6 +111,13 @@ const handleToggleWatchMode = (item?: SidebarItem) => {
     item.watchMode = !item.watchMode
   }
 }
+
+const selectedResultId = computed(() => {
+  if (selectedSearchResult.value === -1) return undefined
+  const result =
+    searchResultsWithPlaceholderResults.value[selectedSearchResult.value]
+  return result ? `#search-input-${result.item.id}` : undefined
+})
```
Comment (amritk) [inline packages/api-client/src/views/Request/RequestSidebar.vue:120]:
```suggestion
const selectedResultId = computed(() => {
  const result =
    searchResultsWithPlaceholderResults.value[selectedSearchResult.value]
  return result ? `#search-input-${result.item.id}` : undefined
})
```

Comment (amritk) [reply packages/api-client/src/views/Request/RequestSidebar.vue:120]: this should work but not sure if that new typescript rule will complain

**Review by amritk — APPROVED** [review]
Pretty cool I can tab around the whole app! 

when vim bindings?

there is this one error though
image

**Comment by hwkr** [comment]
@amritk Weird error fixed 🫡

**Comment by amritk** [comment]
Send ittt

File: packages/api-client/src/layouts/Modal/ApiClientModal.vue [hunk 3]
```diff
@@ -46,14 +71,21 @@ onBeforeUnmount(() => {
 <template>
   <div
     v-show="modalState.open"
-    class="scalar">
-    <div className="scalar-container z-overlay">
-      <div className="scalar-app scalar-client scalar-app-layout">
+    class="scalar scalar-app">
+    <div class="scalar-container z-overlay">
+      <div
+        :id="id"
```
Comment (amritk) [inline packages/api-client/src/layouts/Modal/ApiClientModal.vue:77]: @hwkr just checking what this ID was for? Do we need it for anything?

Comment (antlio) [reply packages/api-client/src/layouts/Modal/ApiClientModal.vue:77]: set for the [focus fallback](https://github.com/scalar/scalar/pull/3754/files/ea2f7c88ef030fbb8a3faf50955ce4765f3a6a39#diff-1591f5ca0e095568e55315761457210130b8db255bc2a97e8cb4a69c5d22fb3cR25)

Comment (amritk) [reply packages/api-client/src/layouts/Modal/ApiClientModal.vue:77]: anthony out of nowhere! ty

**Pull request merged**